### PR TITLE
Setting ruler.evaluation-delay-duration to be deprecated. 

### PR DESCRIPTION
**Pull request by danielblando**
**What this PR does**:
Follow up from https://github.com/cortexproject/cortex/pull/6085

Deprecating `-ruler.evaluation-delay-duration`
Should use new flag `ruler.query-offset`

**Checklist**
- [X] Tests updated
- [ ] Documentation added
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
Change looks great!
Integration test for ruler failed on `TestRulerEvaluationDelay`. Let's just remove that test.

**Review by mustafain117 — APPROVED** [review]

**Comment by rapphil** [comment]
Was this flag experimental? If no, do we need to give a grace period before removing altogether and have just the warning message for now if this flag is being used?

**Comment by yeya24** [comment]
https://cortexmetrics.io/docs/configuration/v1guarantees/#flags-config-and-minor-version-upgrades

I am trying to understand this process better.

```
Upgrading cortex from one minor version to the next should “just work”; that being said, we don’t want to bump the major version every time we remove a flag, so we will will keep deprecated flags around for 2 minor release. There is a metric (deprecated_flags_inuse_total) you can alert on to find out if you’re using a deprecated flag.
```

`just work` means we need to keep the same functionality or there shouldn't be any issue to run Cortex with the existing config?

If the latter this PR should be ok.

**Comment by danielblando** [comment]
I think the biggest issue here is that now we have two flags doing the same thing. Looking at the code it looks like the delay would be the sum of two values. This seems more like a bug than a flag deprecate. We can go back to remove the new RuleQueryOffset flag and maintain RulerEvaluationDelay, but now as the query-offset time for Prometheus.

**Comment by yeya24** [comment]
@danielblando I would prefer to change this flag as a noop now. As keeping the functionality of this flag would break the rule level query offset as you mentioned.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 * [CHANGE] Querier: Deprecate and enable by default `querier.ingester-metadata-streaming` flag. #6147
 * [CHANGE] QueryFrontend/QueryScheduler: Deprecate `-querier.max-outstanding-requests-per-tenant` and `-query-scheduler.max-outstanding-requests-per-tenant` flags. Use frontend.max-outstanding-requests-per-tenant instead. #6146
 * [CHANGE] Ingesters: Enable 'snappy-block' compression on ingester clients by default. #6148
+* [CHANGE] Ruler: Scheduling `evaluation-delay-duration` to be deprecated. Use the highest value between `evaluation-delay-duration` and `ruler.query_offset` #6149
```
Comment (yeya24) [inline CHANGELOG.md:12]: Can we align the name of the flag? Let's use `ruler.evaluation-delay-duration` rather than `evaluation-delay-duration`.

**Review by yeya24 — COMMENTED** [review]
Looks good. Just a small nit.

**Review by yeya24 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * [CHANGE] Querier: Deprecate and enable by default `querier.ingester-metadata-streaming` flag. #6147
 * [CHANGE] QueryFrontend/QueryScheduler: Deprecate `-querier.max-outstanding-requests-per-tenant` and `-query-scheduler.max-outstanding-requests-per-tenant` flags. Use frontend.max-outstanding-requests-per-tenant instead. #6146
 * [CHANGE] Ingesters: Enable 'snappy-block' compression on ingester clients by default. #6148
+* [CHANGE] Ruler: Scheduling `ruler.evaluation-delay-duration` to be deprecated. Use the highest value between `ruler.evaluation-delay-duration` and `ruler.query_offset` #6149
```
Comment (yeya24) [inline CHANGELOG.md:12]: It should be `ruler.query-offset`

**Review by yeya24 — APPROVED** [review]

**Pull request merged**